Source/JavaScriptCore/assembler/MacroAssemblerARM64.h

```diff
@@ -177,8 +177,8 @@ class MacroAssemblerARM64 : public AbstractMacroAssembler<Assembler> {
     {
         if (isUInt12(imm.m_value))
             m_assembler.add<32>(dest, src, UInt12(imm.m_value));
-        else if (isUInt12(-imm.m_value))
-            m_assembler.sub<32>(dest, src, UInt12(-imm.m_value));
+        else if (isUInt12(toTwosComplement(imm.m_value)))
+            m_assembler.sub<32>(dest, src, UInt12(toTwosComplement(imm.m_value)));
```

Comment:
How is this different?  What value does this `toTwosComplement` function add?

Reply:
Added description in commit message.